intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c

```diff
@@ -3133,6 +3119,453 @@ struct ne_tensor* ne_conv_1d_2s(struct ne_context* ctx, struct ne_tensor* a, str
   return result;
 }
 
+static void ne_set_op_params(struct ne_tensor* tensor, const void* params, size_t params_size) {
+  NE_ASSERT(tensor != NULL);  // silence -Warray-bounds warnings
+  // assert(params_size <= NE_MAX_OP_PARAMS);
+  memcpy(tensor->op_params, params, params_size);
+}
+
+// for ne_conv_1d
+static int64_t ne_calc_conv_output_size(int64_t ins, int64_t ks, int s, int p, int d) {
+  return (ins + 2 * p - d * (ks - 1) - 1) / s + 1;
+}
+
+NE_API struct ne_tensor* ne_conv_1d(struct ne_context* ctx, struct ne_tensor* a, struct ne_tensor* b, int s0, int p0,
+                                    int d0) {
+  NE_ASSERT(ne_is_matrix(b));
+  NE_ASSERT(a->ne[1] == b->ne[1]);
+  bool is_node = false;
+
+  if (a->grad || b->grad) {
+    NE_ASSERT(false);  // TODO: implement backward
+    is_node = true;
+  }
+
+  const int64_t ne[4] = {
+      ne_calc_conv_output_size(b->ne[0], a->ne[0], s0, p0, d0),
+      a->ne[2],
+      1,
+      1,
+  };
+  struct ne_tensor* result = ne_new_tensor(ctx, NE_TYPE_F32, 2, ne, NE_SIZE_CALC);
+
+  int32_t params[] = {s0, p0, d0};
+  ne_set_op_params(result, params, sizeof(params));
+
+  result->op = NE_OP_CONV_1D;
+  result->grad = is_node ? ne_dup_tensor(ctx, result) : NULL;
+  result->src0 = a;
+  result->src1 = b;
+
+  return result;
+}
+
+// ne_conv_1d_ph
+
+struct ne_tensor* ne_conv_1d_ph(struct ne_context* ctx, struct ne_tensor* a, struct ne_tensor* b, int s, int d) {
+  return ne_conv_1d(ctx, a, b, s, a->ne[0] / 2, d);
+}
+
+// ne_compute_forward_conv_1d
+
+static void ne_compute_forward_conv_1d_s1_ph_f16_f32(const struct ne_compute_params* params,
+                                                     const struct ne_tensor* src0, const struct ne_tensor* src1,
+                                                     struct ne_tensor* dst) {
```

Comment:
can you separate all conv codes to a single file since jblas will support conv soon?

Reply:
Here only one func,
we can move to a single file when jblas conv is ready.